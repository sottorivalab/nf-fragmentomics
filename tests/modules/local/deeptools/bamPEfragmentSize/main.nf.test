nextflow_process {

    name "Test Process BAMPEFRAGMENTSIZE"
    script "modules/local/deeptools/bamPEfragmentSize/main.nf"
    process "BAMPEFRAGMENTSIZE"

    test("Should run bamPEFragmentSize with default parameters") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should run bamPEFragmentSize with custom parameters") {

        when {
            params {
                outdir = "tests/results"
                args = "--maxFragmentLength 1000 --binSize 10"
            }
            process {
                
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            // FIXME check somewhere else
            assert path(process.out.histogram.get(0).get(1)).text.contains("--maxFragmentLength 1000")
            assert path(process.out.histogram.get(0).get(1)).text.contains("--binSize 10")
        }

    }
    
}
