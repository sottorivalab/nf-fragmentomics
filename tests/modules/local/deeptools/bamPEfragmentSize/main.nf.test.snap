{
    "Should run bamPEFragmentSize with default parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_fragmentsize.png:md5,1607ab8d1c4aeb0076cb177411826f66",
                        "test.paired_end.sorted_fragmentsize.txt:md5,aec545fc634ae0d1df19a1c10bcd1571"
                    ]
                ],
                "bamqc": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_fragmentsize.png:md5,1607ab8d1c4aeb0076cb177411826f66",
                        "test.paired_end.sorted_fragmentsize.txt:md5,aec545fc634ae0d1df19a1c10bcd1571"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T17:13:31.3952"
    },
    "Should run bamPEFragmentSize with custom parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_fragmentsize.png:md5,1607ab8d1c4aeb0076cb177411826f66",
                        "test.paired_end.sorted_fragmentsize.txt:md5,aec545fc634ae0d1df19a1c10bcd1571"
                    ]
                ],
                "bamqc": [
                    [
                        {
                            "caseid": "test",
                            "sampleid": "test",
                            "timepoint": "T1"
                        },
                        "test.paired_end.sorted_fragmentsize.png:md5,1607ab8d1c4aeb0076cb177411826f66",
                        "test.paired_end.sorted_fragmentsize.txt:md5,aec545fc634ae0d1df19a1c10bcd1571"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T17:14:54.754929"
    }
}